### PR TITLE
fix&feat

### PR DESCRIPTION
**Pull request by holytimy4**
refactored and modified main function added new features like validation and others

### CODE REVIEW

File: config_parser.js [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
+        const config = JSON.parse(fs.readFileSync(configPath, 'utf-8'));
+        console.log('Config loaded successfully.');
 
-    if (!valid) {
-        console.log('Validation errors:', validate.errors);
-        throw new Error('Config validation failed');
-    }
+        const finalConfig = { ...defaultValues, ...config };
+
+        const validate = ajv.compile(schema);
```
Comment (VadimOpt) [inline config_parser.js:19]: It's a great idea to make a feature with validation, think of another way to update the function.
photo_2024-06-12_20-24-29

**Review by VadimOpt — COMMENTED** [review]
Keep it up!

File: config_parser.js [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
+        const valid = validate(finalConfig);
+
+        if (!valid) {
+            console.log('Validation errors:', validate.errors);
+            throw new Error('Config validation failed');
+        }
 
-    return config;
+        if (preprocess) {
```
Comment (gfdsokfsdn3) [inline config_parser.js:27]: Great idea, but try to improove this in next commit.

**Review by gfdsokfsdn3 — COMMENTED** [review]
Left my comment below

**Pull request merged**